app/assets/javascripts/components/sampleUpload.jsx

```diff
@@ -0,0 +1,278 @@
+class SampleUpload extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.csrf = props.csrf;
+    this.project = props.project ? props.project : null;
+    this.handleProjectSubmit = this.handleProjectSubmit.bind(this);
+    this.clearError = this.clearError.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.state = {
+      allProjects: props.projects ? props.projects : null,
+      invalid: false,
+      errorMessage: '',
+      success: false,
+      successMessage: '',
+      project: 'Select a Project'
+    }
+  }
+
+  componentDidMount() {
+    this.initializeSelectTag();
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    this.clearError();
+    if(!this.isFormInvalid()) {
+      this.createSample()
+    }
+  }
+
+  initializeSelectTag() {
+    $('select').material_select();
+  }
+
+  clearError() {
+    this.setState({ 
+      invalid: false,
+      success: false
+     })
+  }
+
+  gotoPage(path) {
+    location.href = `${path}`
+  }
+
+  componentDidUpdate() {
+    this.handleProjectSubmit ? this.handleProjectSubmit : null
+  }
+
+  handleProjectSubmit(e) {
+    e.preventDefault();
+    this.clearError();
+    if(!this.isProjectInvalid()) {
+      this.addProject()
+    }
+  }
+
+  addProject() {
+    var that = this;
+    axios.post('/projects.json', {
+      project: {
+        name: this.refs.new_project.value,
+      },
+      authenticity_token: this.csrf
+    })
+    .then((response) => {
+      var newProjectList = that.state.allProjects.slice();
+      newProjectList.push(response.data);
+      that.setState({
+        allProjects: newProjectList,
+        success: true,
+        successMessage: 'Project added successfully'
+      }, () => {
+        this.refs.new_project.value = '';
+        that.initializeSelectTag();
+      });
+    })
+    .catch((error) => {
+      that.setState({
+        invalid: true,
+        errorMessage: 'Project exists already or is invalid'
+      })
+    });
+  }
+
+  isProjectInvalid() {
+    if (this.refs.new_project.value === '') {
+      this.setState({
+        invalid: true,
+        errorMessage: 'Please enter valid project name'
+      })
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  createSample() {
+    var that = this;
+    axios.post('/samples', {
+      sample: {
+        name: this.refs.name.value.trim(),
+        project_name: this.state.project.trim(),
+        input_files_attributes: [{source_type: 's3', source: this.refs.first_file_source.value.trim() },
+        {source_type: 's3', source: this.refs.second_file_source.value.trim() }],
+        s3_preload_result_path: this.refs.s3_preload_result_path.value.trim(),
+        job_queue: this.refs.job_queue.value.trim(),
+        memory: this.refs.memory.value.trim(),
+        status: 'created'
```

Comment:
I don't think you should specify ```status: 'created'``` here, that should be set when the post actually succeeds -- @yunfangjuan ?

Reply:
you do need that actually. the status needs to be set to created to trigger all the following actions

Reply:
ok thanks for clarifying